pkg/resources/diff_suppressions_test.go

```diff
@@ -204,3 +204,86 @@ func Test_ignoreNewEmptyList(t *testing.T) {
 		})
 	}
 }
+
+func Test_IgnoreMatchingColumnNameAndMaskingPolicyUsingFirstElem(t *testing.T) {
+	resourceSchema := map[string]*schema.Schema{
+		"column": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"column_name": {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+					"masking_policy": {
+						Type: schema.TypeList,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"using": {
+									Type:     schema.TypeList,
+									Optional: true,
+									Elem: &schema.Schema{
+										Type: schema.TypeString,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	resourceData := func(using ...any) map[string]any {
+		return map[string]any{
+			"column": []any{
+				map[string]any{
+					"column_name": "foo",
+					"masking_policy": []any{
+						map[string]any{
+							"using": using,
+						},
+					},
+				},
+			},
+		}
+	}
+	tests := []struct {
+		name         string
+		key          string
+		old          string
+		new          string
+		resourceData *schema.ResourceData
+		wantSuppress bool
+	}{
+		// TODO: add more cases?
```

Comment:
Looking at the function and those tests I'm not sure what it should do, let's talk today.

Reply:
+1

Reply:
Finally understand it, but let's leave some kind of follow-up ticket to improve this. Currently, It's hard to read and understand.

Reply:
Added a follow-up SNOW-1852423.